### PR TITLE
Add html message with files and directories.

### PR DESCRIPTION
**Pull request by k-owens**
Commit leaves a dependency, which will need to be fixed in a later commit.

### CODE REVIEW

File: HTTPServer.test/ServerTests.cs [hunk 1]
```diff
@@ -91,36 +91,89 @@ public void ServerCanReturn404()
...
         [TestMethod]
         public void ServerWillGive400ForWrongVersion()
         {
-            TestResponse("GET / HTTP/1.0", "HTTP/1.1 400 Bad Request\r\n");
+            TestResponse("GET / HTTP/1.0", "HTTP/1.1 400 Bad Request\r\n","");
         }
 
-        private static void TestResponse(string request, string expectedReply)
+        [TestMethod]
+        public void ServerCanReturnFilesInDirectory()
```
Comment (aaronlahey) [inline HTTPServer.test/ServerTests.cs:116]: This will only work on your computer right?

File: HTTP Server.core/Server.cs [hunk 2]
```diff
@@ -82,18 +85,40 @@ private  byte[] HandleReply(byte[] request, int requestSize)
...
-                return Encoding.UTF8.GetBytes("HTTP/1.1 404 Not Found\r\n");
+                    return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 200 OK\r\n"));
+                return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 404 Not Found\r\n"));
             }
-            return Encoding.UTF8.GetBytes("HTTP/1.1 400 Bad Request\r\n");
+            return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 400 Bad Request\r\n"));
         }
 
-        private bool isRequestMethod(string str)
+        public string CreateMessage(string str)
```
Comment (aaronlahey) [inline HTTP Server.core/Server.cs:99]: Once you're adding a body you need to indicate how many bytes are in it using the `Content-Length` header.

File: HTTPServer.test/ServerTests.cs (hunk 1 above)
Comment (k-owens) [inline HTTPServer.test/ServerTests.cs:116]: Yes.  I plan to fix that in the next pull request.

File: HTTP Server.core/Server.cs (hunk 2 above)
Comment (k-owens) [inline HTTP Server.core/Server.cs:99]: Got it.  I didn't realize that it needed to be there already.

File: HTTPServer.driver/Driver.cs [hunk 3]
```diff
@@ -9,29 +9,43 @@ namespace HTTPServer.driver
...
         {
-            server.Start(HandlePort(args), new NetworkSocket());
+            HandleCommands(args);
+            server.Start(port, new NetworkSocket(), directoryPath);
             server.HandleClients();
 
         }
 
-        private static int HandlePort(string[] args)
+        private static void HandleCommands(string[] args)
```
Comment (aaronlahey) [inline HTTPServer.driver/Driver.cs:23]: There must be a better/cleaner way to parse these arguments than a nested if/else.

File: HTTP Server.core/Server.cs [hunk 4]
```diff
@@ -10,10 +11,12 @@ public class Server
         public bool Running = false;
         private ISocket socket;
         private ISocket clientConnection;
+        private string directoryPath;
 
-        public Server Start(int port, ISocket startSocket)
+        public Server Start(int port, ISocket startSocket, string directory)
```
Comment (aaronlahey) [inline HTTP Server.core/Server.cs:16]: What if there's another option you need to pass to the server in the future? Can you structure the code such that the server constructor doesn't need to change when it needs more configuration?

File: HTTPServer.test/ServerTests.cs [hunk 5]
```diff
@@ -25,14 +25,14 @@ public class ServerUnitTests
         [TestMethod]
         public void ServerCanStart()
         {
-            Assert.True(server.Start(8080, new NetworkSocket()) != null && server.Running);
+            Assert.True(server.Start(8080, new NetworkSocket(), "") != null && server.Running);
```
Comment (aaronlahey) [inline HTTPServer.test/ServerTests.cs:28]: It's strange to me that `server.start()` would return the server or `null`. Returning `null` is still you indicating some sort of error condition. I don't think starting a server which is already started should be an error.

File: HTTP Server.core/Server.cs [hunk 6]
```diff
@@ -82,18 +85,40 @@ private  byte[] HandleReply(byte[] request, int requestSize)
...
+                return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 404 Not Found\r\n"));
             }
-            return Encoding.UTF8.GetBytes("HTTP/1.1 400 Bad Request\r\n");
+            return Encoding.UTF8.GetBytes(CreateMessage("HTTP/1.1 400 Bad Request\r\n"));
         }
 
-        private bool isRequestMethod(string str)
+        public string CreateMessage(string str)
+        {
+            if (directoryPath.Equals(""))
```
Comment (aaronlahey) [inline HTTP Server.core/Server.cs:101]: I don't like that the path is a string. There must be richer types to use.

**Comment by aaronlahey** [comment]
This doesn't look finished to me. There are a lot of magic strings, the server has many responsibilities (socket connection, reading files, formatting html, routing, building responses, etc). It seems to have a lot of primitive obsession as well. The directory is a string, responses are strings, methods are strings... It looks rushed. I'd like to see a cohesive PR which completely implements a feature a represents work that could merged into master and potentially deployed; not work which leaves master in an in-between state or prevents other developers from working on it.

**Pull request closed without merging**